plugins/SlicerT/SlicerTView.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * SlicerTView.h - declaration of class SlicerTView
+ *
+ * Copyright (c) 2023 Daniel Kauss Serna <daniel.kauss.serna@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_SLICERT_VIEW_H
+#define LMMS_SLICERT_VIEW_H
+
+#include <QPushButton>
+
+#include "ComboBox.h"
+#include "Instrument.h"
+#include "InstrumentView.h"
+#include "Knob.h"
+#include "LcdSpinBox.h"
+#include "LedCheckBox.h"
+#include "PixmapButton.h"
+#include "SlicerTWaveform.h"
+
+namespace lmms {
+
+class SlicerT;
+
+namespace gui {
+
+class SlicerTView : public InstrumentViewFixedSize
+{
+	Q_OBJECT
+
+protected slots:
+	void exportMidi();
```

Comment:
Because it is a Qt slot, and gets connected to the button. I think it has to be a slot for this? I'm not really sure, but everywhere else it is done like this.

Reply:
> Because it is a Qt slot, and gets connected to the button. I think it has to be a slot for this? I'm not really sure, but everywhere else it is done like this.

Qt is now very loose when it comes to putting slots in a slot section. You can pass in regular functions and even lambdas. Signals generally still need to be put under the `signals` section I believe, but since this is a slot we are talking about, you can probably just remove the `protected slots` part and move the function to be private. My question was mostly about why it's protected though; it doesn't seem like it needs to be as such.

Reply:
> Qt is now very loose when it comes to putting slots in a slot section. You can pass in regular functions and even lambdas. Signals generally still need to be put under the signals section I believe, but since this is a slot we are talking about, you can probably just remove the protected slots part and move the function to be private. My question was mostly about why it's protected though; it doesn't seem like it needs to be as such.

Ok, but I would still leave it as a slot for readability. I don't really have a reason as to why it was protected, its probably some leftover from when I started. I'll just make it a public slot.